File: android-app/app/src/main/java/com/chitui/webview/MainActivity.java
package com.chitui.webview;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private static final String PREFS_NAME = "ChitUIPrefs";
    private static final String SERVER_URL_KEY = "server_url";

    private EditText serverAddressInput;
    private Button connectButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Check if server address is already saved
        SharedPreferences prefs = getSharedPreferences(PREFS_NAME, MODE_PRIVATE);
        String savedUrl = prefs.getString(SERVER_URL_KEY, null);

        if (savedUrl != null && !savedUrl.isEmpty()) {
            // Server address exists, go directly to WebView
            openWebView(savedUrl);
            finish();
            return;
        }

        // Show server address input screen
        setContentView(R.layout.activity_main);

        serverAddressInput = findViewById(R.id.server_address_input);
        connectButton = findViewById(R.id.connect_button);

        connectButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String serverAddress = serverAddressInput.getText().toString().trim();

                if (TextUtils.isEmpty(serverAddress)) {
                    Toast.makeText(MainActivity.this, "Please enter a server address", Toast.LENGTH_SHORT).show();
                    return;
                }

                // Add http:// if no protocol specified
                if (!serverAddress.startsWith("http://") && !serverAddress.startsWith("https://")) {
                    serverAddress = "http://" + serverAddress;
                }

                // Save server address
                SharedPreferences.Editor editor = prefs.edit();
                editor.putString(SERVER_URL_KEY, serverAddress);
                editor.apply();

                // Open WebView
                openWebView(serverAddress);
                finish();
            }
        });
    }

    private void openWebView(String url) {
        Intent intent = new Intent(this, WebViewActivity.class);
        intent.putExtra("url", url);
        startActivity(intent);
    }
}

File: ChitUIApp/app/src/main/java/com/chitui/app/MainActivity.kt
package com.chitui.app

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.chitui.app.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    companion object {
        private const val PREFS_NAME = "ChitUIPrefs"
        private const val SERVER_URL_KEY = "server_url"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Check if server address is already saved
        val prefs = getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        val savedUrl = prefs.getString(SERVER_URL_KEY, null)

        if (!savedUrl.isNullOrEmpty()) {
            // Server address exists, go directly to WebView
            openWebView(savedUrl)
            finish()
            return
        }

        // Show server address input screen
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.connectButton.setOnClickListener {
            val serverAddress = binding.serverAddressInput.editText?.text.toString().trim()

            if (serverAddress.isEmpty()) {
                Toast.makeText(this, "Please enter a server address", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            // Add http:// if no protocol specified
            val finalUrl = if (!serverAddress.startsWith("http://") &&
                             !serverAddress.startsWith("https://")) {
                "http://$serverAddress"
            } else {
                serverAddress
            }

            // Save server address
            prefs.edit().putString(SERVER_URL_KEY, finalUrl).apply()

            // Open WebView
            openWebView(finalUrl)
            finish()
        }
    }

    private fun openWebView(url: String) {
        val intent = Intent(this, WebViewActivity::class.java).apply {
            putExtra("url", url)
        }
        startActivity(intent)
    }
}
